src/Nethermind/Nethermind.Synchronization/FastBlocks/FastHeadersSyncFeed.cs

```diff
@@ -44,11 +44,8 @@ public class HeadersSyncFeed : ActivatedSyncFeed<HeadersSyncBatch?>
 
         private readonly ulong _fastHeadersMemoryBudget;
         protected long _lowestRequestedHeaderNumber;
-
-        protected Hash256 _nextHeaderHash;
-        protected UInt256? _nextHeaderTotalDifficulty;
-
         protected long _pivotNumber;
+        protected (Hash256 Hash, UInt256? TotalDifficulty) _expectedNextHeader;
```

Comment:
This is a complex struct, I think it's assignment isn't guaranteed to be atomic. If you want atomic `=` than make it a reference type.

Reply:
Changed to record.